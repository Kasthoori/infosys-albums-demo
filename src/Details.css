.SuperContainer {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.TopBanner {
  background-color: #bdbdbd;
  border-radius: 10px;
  width: 90%;
  margin-top: 10px;
  padding-left: 90px;
}

.MainContainer {
  width: 95%;
  display: flex;
  flex-direction: row;
  /* padding-left: 100px; */
  /* padding-right: 20px; */
  justify-content: center;
}

.List {
  width: 40%;
}

.LeftView {
  /* border: 1px solid black; */
  width: 50%;
  overflow: hidden;
  display: flex;
  /* justify-content: center; */
  align-items: center;
  margin-top: 20px;
  flex-direction: column;
}

#albumimg {
  width: 400px;
  height: 400px;
  margin-top: 1px;
}

#idtext {
  font-size: 20px;
  font-weight: bold;
  margin-top: -10px;
}

#titletext {
  font-size: 18px;
  font-weight: bold;
  margin-top: 5px;
  color: grey;
}

#aid {
  margin-top: -15px;
}

#uid {
  margin-top: -10px;
}
