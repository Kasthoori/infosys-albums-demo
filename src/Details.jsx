import React, { useState, useEffect } from "react";
import { useLocation, useHistory } from "react-router-dom";
import ListItemImg from "./componenets/ListItemImg/ListItemImg";
import "./Details.css";

const Details = () => {
  const location = useLocation();
  const history = useHistory();

  let status = "";
  let albumtitle = "";
  let userId = "";
  let aId = "";
  let tracks = [];

  const [albumpic, setAlbumPic] = useState("");
  const [albumId, setAlbumId] = useState("");
  const [id, setId] = useState("");
  const [title, setTitle] = useState("");

  try {
    status = location.state.status;
    albumtitle = location.state.title;
    userId = location.state.userId;
    tracks = location.state.tracks;
    aId = location.state.id;
  } catch (err) {
    status = "false";
    tracks = [];
  }

  useEffect(() => {
    if (status !== "true") {
      history.push("/");
    }

    if (tracks.length > 0) {
      setAlbumPic(tracks[0].url);
      setAlbumId(tracks[0].albumId);
      setTitle(tracks[0].title);
      setId(tracks[0].id);
      console.log("YES");
    }
  }, [history, status, tracks]);

  const listItemImgOnClick = (track) => {
    setAlbumPic(track.url);
    setAlbumId(track.albumId);
    setId(track.id);
    setTitle(track.title);
  };

  return (
    <div className="SuperContainer">
      <div className="TopBanner">
        <h1>Album Title: {albumtitle}</h1>
        <h2 id="aid">Album Id: {aId}</h2>
        <h3 id="uid">User Id: {userId}</h3>
      </div>

      <div className="MainContainer">
        <div className="List">
          {tracks.map((track) => (
            <ListItemImg track={track} onclick={listItemImgOnClick} />
          ))}
        </div>

        <div className="LeftView">
          <img id="albumimg" src={albumpic} alt="" />

          <h1>Album ID: {albumId}</h1>
          <span id="idtext">ID: {id}</span>
          <span id="titletext">Title: {title} </span>
        </div>
      </div>
    </div>
  );
};

export default Details;
