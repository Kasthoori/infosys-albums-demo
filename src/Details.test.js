/* eslint-disable no-undef */
import React, { useState, useEffect } from "react";
import { shallow } from "enzyme";
import { cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";

import "@testing-library/jest-dom/extend-expect";
import toJson from "enzyme-to-json";

import Details from "./Details";

jest.mock("react-router-dom", () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: "",
    search: "",
    hash: "",
    state: null,
    key: "",
  }),
  useHistory: () => {
    push: jest.fn();
  },
}));

describe("Testing Details", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Details />);
  });

  afterEach(cleanup);

  it("Details render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
