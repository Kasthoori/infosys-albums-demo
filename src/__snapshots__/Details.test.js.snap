// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Details Details render without crashing 1`] = `ShallowWrapper {}`;
